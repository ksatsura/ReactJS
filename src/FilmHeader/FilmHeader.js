//imports
import * as React from 'react';
import FilmInfo from '../FilmInfo/FilmInfo';
import Poster from '../Poster/Poster';
import { connect } from 'react-redux';
import { backToSearch } from '../actions/filmActions';
import CompanyTitle from '../CompanyTitle/CompanyTitle';
import { Link }  from 'react-router-dom';
import '../style.css';

// @flow
type Props = {
    film: {
        poster_path: string,

    },
    handleClick: () =>  void,
    filmsSameGenre: Array<{}>,
    value: string
};

export const FilmHeader = (props: Props) => {

    const { film, handleClick, value } = props;

    return (
        <div className='film-info'>
            <div className='top-film-info'>
                <CompanyTitle />
                <Link to={`/search/${value}`}>
                    <button onClick={handleClick} className='search-navigation'>search</button>
                </Link>
            </div>
            <div className='film-poster-data'>
                <div className='poster'>
                    <Poster url={film.poster_path}/>
                </div>
                <FilmInfo film={film} />
            </div>
        </div>
    );
};

const mapStateToProps = state => {

    return {
        film: state.filmReducer.film,
        value: state.filmListReducer.currentValue,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        handleClick: () => { dispatch(backToSearch()); },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(FilmHeader);