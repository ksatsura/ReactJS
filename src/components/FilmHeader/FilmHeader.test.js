import React from 'react';
import { FilmHeader } from './FilmHeader';
import {shallow} from 'enzyme';

describe('FilmHeader.js', () => {

    const props = {
        film: {
            poster_path: 'src/images/wallpaper.jpg',
        },
        handleClick: jest.fn(),
    };

    describe('should renders', () => {

        it('as expected', () => {
            const component = shallow(<FilmHeader {...props} />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('handleClick', () => {

        it('test search button click event', () => {
        
            const component = shallow(<FilmHeader {...props} />);
    
            component.find('button').simulate('click');
            expect(props.handleClick).toHaveBeenCalledTimes(1);
        });

    });
});
