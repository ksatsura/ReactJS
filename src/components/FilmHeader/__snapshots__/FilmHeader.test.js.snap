// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmHeader.js should renders as expected 1`] = `
<div
  className="film-info"
>
  <div
    className="top-film-info"
  >
    <Ð¡ompanyTitle />
    <button
      className="search-navigation"
      onClick={[MockFunction]}
    >
      search
    </button>
  </div>
  <div
    className="film-poster-data"
  >
    <div
      className="poster"
    >
      <Poster
        url="src/images/wallpaper.jpg"
      />
    </div>
    <FilmInfo
      film={
        Object {
          "poster_path": "src/images/wallpaper.jpg",
        }
      }
    />
  </div>
</div>
`;
